### PR TITLE
fix/ShadowAccountManager static state

### PR DESCRIPTION
**Pull request by Devashishbasu**
Making instance variable to static variable of ShadowAccountManager

### CODE REVIEW

**Comment by Devashishbasu** [comment]
Hi @utzcoz @hoisie , could you please review

File: integration_tests/ctesque/src/androidTest/java/android/app/ActivityInstrTest.java [hunk 1]
```diff
@@ -113,4 +115,44 @@ public void audioManager_activityContextEnabled() {
...
+
+  @Test
+  public void accountManager_activityContextEnabled() {
+    String originalProperty = System.getProperty("robolectric.createActivityContexts", "");
+    System.setProperty("robolectric.createActivityContexts", "true");
+    try {
+      AccountManager applicationAccountManager =
+          (AccountManager)
+              ApplicationProvider.getApplicationContext().getSystemService(Context.ACCOUNT_SERVICE);
+      Log.d("Test", "Retrieved applicationAccountManager instance");
```
Comment (hoisie) [inline integration_tests/ctesque/src/androidTest/java/android/app/ActivityInstrTest.java:127]: Remove these log statements, they are not necessary. The assertions will check the values.

File: integration_tests/ctesque/src/androidTest/java/android/app/ActivityInstrTest.java [hunk 2]
```diff
@@ -113,4 +115,44 @@ public void audioManager_activityContextEnabled() {
...
+              Account[] applicationAccounts =
+                  applicationAccountManager.getAccountsByType("com.example.account_type");
+              Account[] activityAccounts =
+                  activityAccountManager.getAccountsByType("com.example.account_type");
+
+              Log.d("TestActivity", "applicationAccounts length: " + applicationAccounts.length);
+              Log.d("TestActivity", "activityAccounts length: " + activityAccounts.length);
+
+              // Add assertions to validate behavior
+              assertThat(activityAccounts.length).isEqualTo(applicationAccounts.length);
```
Comment (hoisie) [inline integration_tests/ctesque/src/androidTest/java/android/app/ActivityInstrTest.java:147]: Can this just be: `assertThat(activityAccounts).isEqualTo(applicationAccounts)`?

File: robolectric/src/test/java/org/robolectric/shadows/ShadowAccountManagerTest.java [hunk 3]
```diff
@@ -147,8 +150,9 @@ public void testAddAccountExplicitly_noPasswordNoExtras() {
 
     try {
       am.addAccountExplicitly(null, null, null);
-      fail("An illegal argument exception should have been thrown when trying to add a null"
-               + " account");
+      fail(
```
Comment (hoisie) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowAccountManagerTest.java:153]: Can you do the google-java-format formatting in a separate PR?

File: testapp/src/main/java/org/robolectric/testapp/TestActivity.java [hunk 4]
```diff
@@ -26,4 +29,19 @@ private void setupAudioManager() {
       Log.d("TestActivity", "AudioManager is null");
     }
   }
+
+  private void setupAccountManager() {
```
Comment (hoisie) [inline testapp/src/main/java/org/robolectric/testapp/TestActivity.java:33]: What is the goal of this code? Is the account manager here used anywhere?

**Comment by Devashishbasu** [comment]
@utzcoz ...so i need to reformat this changes  like you did in your pr for audiomanager?

**Comment by Devashishbasu** [comment]
> @Devashishbasu Yeah. You can follow this PR to refactor this PR. And we can review again.

I need to add integration tests in ContextTest instead of ActivityInstrTest right?

**Pull request closed without merging**